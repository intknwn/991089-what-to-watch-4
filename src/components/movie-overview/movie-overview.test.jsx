import React from 'react';
import renderer from 'react-test-renderer';
import MovieOverview from './movie-overview.jsx';

const movie = {
  name: `Die Hard`,
  previewImg: `img/die-hard.jpg`,
  backgroundImg: `https://via.placeholder.com/1300x552`,
  posterImg: `https://via.placeholder.com/273x410`,
  previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  genre: `Action`,
  year: 1988,
  rating: 8.2,
  score: 796619,
  description: `An NYPD officer tries to save his wife and several others taken hostage by German terrorists during a Christmas party at the Nakatomi Plaza in Los Angeles.`,
  director: `John McTiernan`,
  cast: [`Bruce Willis`, `Alan Rickman`, `Bonnie Bedelia`],
  runtime: 132,
};

it(`MovieOverview component should renders correctly`, () => {
  const tree = renderer.create(
      <MovieOverview
        movie={movie}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
