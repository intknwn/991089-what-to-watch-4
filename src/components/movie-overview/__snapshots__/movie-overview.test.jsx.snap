// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieOverview component should renders correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.2
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        796619 ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      An NYPD officer tries to save his wife and several others taken hostage by German terrorists during a Christmas party at the Nakatomi Plaza in Los Angeles.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        John McTiernan
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Bruce Willis
        Alan Rickman
        Bonnie Bedelia
      </strong>
    </p>
  </div>,
]
`;
